.vitepress/config.ts

```diff
@@ -25,12 +26,48 @@ export default defineConfig({
   title: 'StackBlitz Docs',
   description:
     'Discover how to use StackBlitz, an online development environment for frontend, Node.js and the JavaScript ecosystem.',
-  head: getHeadTags(process.env),
+  head: [
+    ['link', { rel: 'icon', type: 'image/png', href: `${BASE}img/theme/favicon.png` }],
+    ...getAnalyticsTags(process.env),
+  ],
 
   // Sitemap
   lastUpdated: true,
   sitemap: {
-    hostname: `https://developer.stackblitz.com${BASE}`,
+    hostname: BASE_WITH_ORIGIN,
+  },
+
+  /**
+   * Generate social media metadata tags at build time.
+   * Note: this does not run when using the development server,
+   * so it must be tested by doing a full build with `npm run build`.
+   */
+  transformHead({ pageData, page }) {
+    // Get the raw title and description from frontmatter,
+    // rather than the title which has the site suffix
+    const { title, description, og_image } = pageData.frontmatter;
+    const og_type = page === 'index.md' ? 'website' : 'article';
+
+    // New meta tags to add to the <head>
+    const tags: HeadConfig[] = [];
+
+    // Add opengraph tags
+    tags.push(['meta', { property: 'og:type', content: og_type }]);
+    tags.push(['meta', { property: 'og:title', content: title }]);
+    if (og_image) {
+      const url = `${BASE_WITH_ORIGIN}img/og/${og_image}`;
+      tags.push(['meta', { property: 'og:image', content: url }]);
+    }
+
+    // Add twitter tags
+    tags.push(['meta', { name: 'twitter:site', content: '@StackBlitz' }]);
+    tags.push(['meta', { name: 'twitter:card', content: 'summary_large_image' }]);
+    tags.push(['meta', { name: 'twitter:title', content: title }]);
+    if (description) {
+      tags.push(['meta', { name: 'twitter:description', content: description }]);
+    }
```

Comment:
The logic we had in YAML frontmatter for each page is now done in JS in the `transformHead` function.